config/cmake_ext_mod/HDFUseFortran.cmake

```diff
@@ -178,36 +109,95 @@ else ()
+  CHECK_FORTRAN_FEATURE(c_sizeof_code ${SOURCE_CODE} ${HDF_PREFIX}_FORTRAN_HAVE_C_SIZEOF)
+endif ()
 
-  # Check for F2008 standard intrinsic function STORAGE_SIZE
-  CHECK_FORTRAN_FEATURE(storage_size_code ${STORAGE_SIZE_CODE} ${HDF_PREFIX}_FORTRAN_HAVE_STORAGE_SIZE)
+# Check for F2008 standard intrinsic function STORAGE_SIZE
+READ_SOURCE("PROGRAM PROG_FC_STORAGE_SIZE" "END PROGRAM PROG_FC_STORAGE_SIZE" SOURCE_CODE)
+if (NOT CMAKE_VERSION VERSION_LESS "3.14.0")
+  check_fortran_source_compiles (${SOURCE_CODE} ${HDF_PREFIX}_FORTRAN_HAVE_STORAGE_SIZE SRC_EXT f90)
+else ()
+  CHECK_FORTRAN_FEATURE(storage_size_code ${SOURCE_CODE} ${HDF_PREFIX}_FORTRAN_HAVE_STORAGE_SIZE)
+endif ()
 
-  # Check for F2008 standard intrinsic module "ISO_FORTRAN_ENV"
+# Check for F2008 standard intrinsic module "ISO_FORTRAN_ENV"
+READ_SOURCE("PROGRAM PROG_FC_ISO_FORTRAN_ENV" "END PROGRAM PROG_FC_ISO_FORTRAN_ENV" SOURCE_CODE)
+if (NOT CMAKE_VERSION VERSION_LESS "3.14.0")
+  check_fortran_source_compiles (${SOURCE_CODE} ${HDF_PREFIX}_HAVE_ISO_FORTRAN_ENV SRC_EXT f90)
+else ()
   set (${HDF_PREFIX}_HAVE_ISO_FORTRAN_ENV FALSE)
-  CHECK_FORTRAN_FEATURE(iso_fortran_env_code ${ISO_FORTRAN_ENV_CODE} ${HDF_PREFIX}_HAVE_ISO_FORTRAN_ENV)
+  CHECK_FORTRAN_FEATURE(iso_fortran_env_code ${SOURCE_CODE} ${HDF_PREFIX}_HAVE_ISO_FORTRAN_ENV)
+endif ()
 
+#READ_SOURCE("PROGRAM PROG_FC_REALISNOTDOUBLE" "END PROGRAM PROG_FC_REALISNOTDOUBLE" REALISNOTDOUBLE_CODE)
+set (REALISNOTDOUBLE_CODE
+  "
+       MODULE type_mod
+         INTERFACE h5t
+           MODULE PROCEDURE h5t_real
+           MODULE PROCEDURE h5t_dble
+         END INTERFACE
+       CONTAINS
+         SUBROUTINE h5t_real(r)
+           REAL :: r
+         END SUBROUTINE h5t_real
+         SUBROUTINE h5t_dble(d)
+           DOUBLE PRECISION :: d
+         END SUBROUTINE h5t_dble
+       END MODULE type_mod
+       PROGRAM main
+         USE type_mod
+         REAL :: r
+         DOUBLE PRECISION :: d
+         CALL h5t(r)
+         CALL h5t(d)
+       END PROGRAM main
+  "
+)
+if (NOT CMAKE_VERSION VERSION_LESS "3.14.0")
+  check_fortran_source_compiles (${REALISNOTDOUBLE_CODE} ${HDF_PREFIX}_FORTRAN_DEFAULT_REAL_NOT_DOUBLE SRC_EXT f90)
+else ()
   set (${HDF_PREFIX}_FORTRAN_DEFAULT_REAL_NOT_DOUBLE FALSE)
   CHECK_FORTRAN_FEATURE(realisnotdouble_code ${REALISNOTDOUBLE_CODE} ${HDF_PREFIX}_FORTRAN_DEFAULT_REAL_NOT_DOUBLE)
+endif ()
 
-  #-----------------------------------------------------------------------------
-  # Checks if the ISO_C_BINDING module meets all the requirements
-  #-----------------------------------------------------------------------------
+#-----------------------------------------------------------------------------
+# Checks if the ISO_C_BINDING module meets all the requirements
+#-----------------------------------------------------------------------------
+#READ_SOURCE("PROGRAM PROG_FC_ISO_C_BINDING" "END PROGRAM PROG_FC_ISO_C_BINDING" ISO_C_BINDING_CODE)
+set (ISO_C_BINDING_CODE
```

Comment:
Why aren't we using READ_SOURCE here, too?

Reply:
There is no equivalent in Autotools, this check is for an issue with windows intel <v12.1